src/aatoolbox/datasources/fewsnet/fewsnet.py

```diff
@@ -0,0 +1,227 @@
+            url=url_country_date,
+            zip_filename=zip_filename,
+            output_dir=output_dir_country,
+        )
+    else:
+        country_data = True
+    return country_data
+
+
+def _download_fewsnet_region(
+    date: datetime,
+    region_name: str,
+    region_code: str,
+    output_dir: Path,
+    use_cache: bool,
+) -> bool:
+    """
+    Download fewsnet data that covers `region_name`.
+
+    Parameters
+    ----------
+    date : datetime
+        date for which the data should be downloaded
+        only the year and month part are used
+        this commonly refers to the month of the Current Situation period
+    region_name : str
+        name of the region to which the `iso2` belongs,
+        e.g. "east-africa"
+    region_code : str
+        code that refers to the `region_name,
+        e.g. "EA"
+    output_dir : Path
+        path to dir to which the data should be written
+    use_cache : bool
+        if True, don't download if output_dir already exists
+
+    Returns
+    -------
+    region_data : bool
+        if True, region data for the given date and region name exists
+    """
+    url_region_date = BASE_URL_REGION.format(
+        region_code=region_code,
+        region_name=region_name,
+        YYYY=date.year,
+        MM=date.month,
+    )
+
+    output_dir_region = output_dir / f"{region_code}{date.strftime('%Y%m')}"
+    zip_filename_region = f"{output_dir_region.name}.zip"
+    if not output_dir_region.exists() or use_cache is False:
+        region_data = _download_zip(
+            url=url_region_date,
+            zip_filename=zip_filename_region,
+            output_dir=output_dir_region,
+        )
+    else:
+        region_data = True
+    return region_data
+
+
+def download_fewsnet(
+    date: datetime,
+    iso2: str,
+    region_name: str,
+    region_code: str,
+    output_dir: Union[Path, str],
+    use_cache=True,
+):
+    """
+    Retrieve the raw fewsnet data.
+
+    Depending on the region and date, this data is published per region or
+    per country. This function retrieves the country data
+    if it exists, and else the regional data for `date` and `iso2`.
+
+    Parameters
+    ----------
+    date : datetime
+        date for which the data should be downloaded
+        only the year and month part are used
+        this commonly refers to the month of the Current Situation period
+    iso2 : str
+        iso2 code of the country of interest
+    region_name : str
+        name of the region to which the `iso2` belongs,
+        e.g. "east-africa"
+    region_code : str
+        code that refers to the `region_name,
+        e.g. "EA"
+    output_dir : Path or str
+        path to dir to which the data should be written
+    use_cache : bool
+        if True, don't download if output_dir already exists
+    """
+    # upper case iso2 and region_code since they are upper-cased
+    # in fewsnet's url
+    iso2 = iso2.upper()
+    region_code = region_code.upper()
```

Comment:
So I guess my reasoning was the following (but no worries if you want to keep it this way, it's a small detail): I thought it's better to pass around the unformatted string, otherwise in every function where it gets passed, you need to specify that it must be a lowercase iso3, because the formatting will not be taken care of later. Just how you pass the datetime object around and only format it at the end, rather than passing the month and year. Hope that makes sense.